src/server/services/wmts/qgswmtsgetcapabilities.cpp

```diff
@@ -0,0 +1,760 @@
+          contactAddressElem.appendChild( contactAddressMailElem );
+          contactInfoElem.appendChild( contactAddressElem );
+        }
+        if ( !contactPhone.isEmpty() )
+        {
+          QDomElement contactPhoneElem = doc.createElement( QStringLiteral( "ows:Phone" ) );
+          QDomElement contactVoiceElem = doc.createElement( QStringLiteral( "ows:Voice" ) );
+          QDomText contactVoiceText = doc.createTextNode( contactPhone );
+          contactVoiceElem.appendChild( contactVoiceText );
+          contactPhoneElem.appendChild( contactVoiceElem );
+          contactInfoElem.appendChild( contactPhoneElem );
+        }
+        serviceContactElem.appendChild( contactInfoElem );
+      }
+      serviceElem.appendChild( serviceContactElem );
+    }
+
+    //End
+    return serviceElem;
+  }
+
+  QDomElement getOperationsMetadataElement( QDomDocument &doc, const QgsProject *project, const QgsServerRequest &request )
+  {
+    //ows:OperationsMetadata element
+    QDomElement operationsMetadataElement = doc.createElement( QStringLiteral( "ows:OperationsMetadata" )/*ows:OperationsMetadata*/ );
+
+    //ows:Operation element with name GetCapabilities
+    QDomElement getCapabilitiesElement = doc.createElement( QStringLiteral( "ows:Operation" )/*ows:Operation*/ );
+    getCapabilitiesElement.setAttribute( QStringLiteral( "name" ), QStringLiteral( "GetCapabilities" ) );
+    operationsMetadataElement.appendChild( getCapabilitiesElement );
+
+    //ows:DCP
+    QDomElement dcpElement = doc.createElement( QStringLiteral( "ows:DCP" )/*ows:DCP*/ );
+    getCapabilitiesElement.appendChild( dcpElement );
+    QDomElement httpElement = doc.createElement( QStringLiteral( "ows:HTTP" )/*ows:HTTP*/ );
+    dcpElement.appendChild( httpElement );
+
+    //Prepare url
+    QString hrefString = serviceUrl( request, project );
+
+    //ows:Get
+    QDomElement getElement = doc.createElement( QStringLiteral( "ows:Get" )/*ows:Get*/ );
+    getElement.setAttribute( QStringLiteral( "xlink:href" ), hrefString );
+    httpElement.appendChild( getElement );
+
+    //ows:Operation element with name GetTile
+    QDomElement getTileElement = getCapabilitiesElement.cloneNode().toElement();//this is the same as 'GetCapabilities'
+    getTileElement.setAttribute( QStringLiteral( "name" ), QStringLiteral( "GetTile" ) );
+    operationsMetadataElement.appendChild( getTileElement );
+
+    //ows:Operation element with name GetFeatureInfo
+    QDomElement getFeatureInfoElement = getCapabilitiesElement.cloneNode().toElement();//this is the same as 'GetCapabilities'
+    getFeatureInfoElement.setAttribute( QStringLiteral( "name" ), QStringLiteral( "GetFeatureInfo" ) );
+    operationsMetadataElement.appendChild( getFeatureInfoElement );
+
+    // End
+    return operationsMetadataElement;
+  }
+
+  QDomElement getContentsElement( QDomDocument &doc, QgsServerInterface *serverIface, const QgsProject *project )
```

Comment:
Can you split this function (450 lines long more or less). It'll be easier to maintain and much more readable :).

Reply:
The method has been splitted